menu.go

```diff
@@ -1,21 +1,110 @@
 package exoskeleton
 
 import (
-	_ "embed"
+	"bytes"
 	"fmt"
 	"sort"
 	"strings"
+	"text/template"
 )
 
+const menuTemplate = "\033[1m" + `USAGE` + "\033[0m" + `
+   {{.Usage}}
+
+{{- range .Sections}}
+
+` + "\033[1m" + `{{.Heading}}` + "\033[0m" + `
+{{- range .MenuItems}}
+   {{rpad .Name .Width}}  {{.Summary}}
+{{- end}}
+{{- end}}
+
+Run ` + "\033[96m" + `{{.HelpUsage}} <command>` + "\033[0m" + ` to print information on a specific command.`
+
+var templateFuncs = template.FuncMap{
+	"rpad": func(s string, padding int) string { return fmt.Sprintf("%*s", -padding, s) },
+}
+
+// SummaryFunc is a function that is expected to return the heading
 type SummaryFunc func(Command) (string, error)
 
-type buildMenuOptions struct {
-	Depth      int
+// MenuOptions are the options that control how menus are constructed for modules.
+type MenuOptions struct {
+	// Depth describes how recursively a menu should be constructed. Its default
+	// value is 0, which indicates that the menu should list only the commands
+	// that are descendants of the module. A value of 1 would list descendants one
+	// level deep, a value of 2 would list descendants two levels deep, etc. A value
+	// -1 lists all descendants.
+	Depth int
+
+	// HeadingFor accepts a Command the Module the menu is being prepared for
+	// and returns a string to use as a section heading for the Command.
+	// The default function returns "COMMANDS".
 	HeadingFor MenuHeadingForFunc
+
+	// SummaryFor accepts a Command and returns its summary and, optionally, an error.
+	// The default function invokes Summary() on the provided Command.
 	SummaryFor SummaryFunc
+
+	// Template is executed with the constructed exoskeleton.Menu to render
+	// help content for a Module.
+	Template *template.Template
 }
 
-func buildMenu(m Module, opts *buildMenuOptions) (*menu, []error) {
+// Menu is the data passed to MenuOptions.Template when it is executed.
+type Menu struct {
+	Usage     string
+	HelpUsage string
+	Sections  MenuSections
+}
+
+type MenuSections []MenuSection
+
+type MenuSection struct {
+	Heading   string
+	MenuItems MenuItems
+}
+
+type MenuItems []*MenuItem
+
+// implement sort.Interface so that MenuItems can be sorted by Name
+
```

Comment:
nit: remove this space?

Reply:
Oh! I put the space in so that "implement sort.Interface so that MenuItems can be sorted by Name" wouldn't be the docstring for `Len()`